### PR TITLE
Fix warnings around Singleton instance method redefinition

### PR DESCRIPTION
**Pull request by carbonin**
Calling .__init__ on Singleton directly was redefining the .instance method
for the singleton class.
This change calls the private method #refresh on the FSTab instance rather
than attempting to reinitialize the instance.

### CODE REVIEW

**Comment by Fryguy** [comment]
Should the clearing of the singleton cache be done higher as it was previously but just in a different way?

**Comment by carbonin** [comment]
The way it was done before didn't actually call the `#refresh` method so it didn't need to be stubbed out.  If I added it as a `before` I would need to stub there also and because I might be using `instance` for the first time I would need to stub multiple calls which could eat up the calls to `File.read` we are `expect`ing in the tests.

File: spec/fstab_spec.rb [hunk 1]
```diff
@@ -1,18 +1,14 @@
...
-  end
-
   it "has newline, single spaces, tab" do
     fstab = <<eos
 
   
 	
 eos
-    expect(File).to receive(:read).with('/etc/fstab').and_return(fstab)
+    expect(File).to receive(:read).at_least(:once).with('/etc/fstab').and_return(fstab)
```
Comment (bdunne) [inline spec/fstab_spec.rb]: This should only be read once, right?

File: spec/fstab_spec.rb [hunk 2]
```diff
@@ -1,18 +1,14 @@
 require 'stringio'
 
 describe LinuxAdmin::FSTab do
-  before do
```
Comment (bdunne) [inline spec/fstab_spec.rb]: @carbonin I think this before should be changed to `subject { described_class.dup }`.  Than you can just change `LinuxAdmin::FSTab` to `subject` and the warnings should be gone.

File: spec/fstab_spec.rb (hunk 1 above)
Comment (carbonin) [inline spec/fstab_spec.rb]: That was to account for if it was the first time we referenced instance and also when we call refresh

**Comment by bdunne** [comment]
:+1:

**Pull request merged**